app-api/routes/dataset.py

```diff
@@ -270,6 +281,11 @@ def list_datasets(
         elif kind == DatasetKind.ANY:
             query = query.filter(or_(Dataset.is_public, Dataset.user_id == user_id))
 
+        # if q is provided, filter by name
+        if q:
+            q = f"%{q}%"
+            query = query.filter(Dataset.name.ilike(q))
+
```

Comment:
allows us to filter list of datasets to return (relevant for the new DatasetSelector)